detector/muonSystem/muonSystemMuRWELL_o1_v01.cpp

```diff
@@ -0,0 +1,941 @@
+
+  //  double endcapDetectorEnvX = endcapDetectorLayerOuterRadius;  // it depends on outer radius
+  //  double endcapDetectorEnvY = endcapDetectorLayerOuterRadius;  // outer radius too
+  double endcapDetectorEnvZ = detectorVolumeThickness;  // layer thickness
+
+  //  double sideY = (numCopiesY -1) * 2 * dimensions.y() + remainderYLength - totalOverlapValueY;
+  //  double sideZ = (numCopiesZ -1) * 2 * dimensions.z() + remainderZLength - totalOverlapValueZ;
+
+  //  int nameCounter = 0;
+  int barrelIdCounter = 1;
+  int endcapIdCounter = 1;
+
+// ----------------------------------------------------------------------------------------------------
+//                               // B A R R E L //
+// ----------------------------------------------------------------------------------------------------
+
+  if (numBarrelDetectorLayers > 0){
+
+  int numBarrelLayer = 1;
+  double barrelLayerRMin = radius - detectorVolumeThickness/2.0;   // have to be changed depends on xml impementation
+  double barrelLayerRMax = radius + 1.5 * detectorVolumeThickness;
+
+  dd4hep::PolyhedraRegular  BarrelDetectorLayer(numSides, barrelLayerRMin, barrelLayerRMax, barrelLength);
+  dd4hep::Volume BarrelDetectorLayerVolume(" barrelDetectorLayer" + numBarrelLayer , BarrelDetectorLayer, mat);
+
+
+  for (int side = 0; side < numSides; ++side) {
+
+  //    dd4hep::Trapezoid sideEnvelope(barrelLength/2.0, barrelLength/2.0, sideTRDLength1/2.0, sideTRDLength2/2.0, detectorVolumeThickness/2.0);      
+      dd4hep::Box sideEnvelope(sideEnvX, sideEnvY, sideEnvZ);
+      dd4hep::Box sideEnvelope2(sideEnvX, sideEnvY2, sideEnvZ);      
+      std::string sideName = dd4hep::xml::_toString(side, "side%d");
+      dd4hep::Volume sideVol(sideName, sideEnvelope, mat);
+      dd4hep::Volume sideVol2(sideName, sideEnvelope2, mat);      
+
+      double angle_degrees = shapeAngle * side; // Calculate the angle for each side
+      double angle_radians = angle_degrees * M_PI / 180.0;
+
+      double sideXOffset = radius * std::cos(angle_radians+shapeAngle_radians);
+      double sideYOffset = radius * std::sin(angle_radians+shapeAngle_radians);
+
+      double sideXOffset2 = (radius + detectorVolumeThickness) * std::cos(angle_radians+shapeAngle_radians);
+      double sideYOffset2 = (radius + detectorVolumeThickness) * std::sin(angle_radians+shapeAngle_radians);      
+
+      dd4hep::RotationZ sideRotationZ(angle_radians+shapeAngle_radians+angle_clearance);
+  //    dd4hep::RotationY sideRotationY(90.* dd4hep::degree);     
+      dd4hep::Rotation3D sideRotation = dd4hep::Rotation3D(sideRotationZ);      
+
+      double sideXPos = sideXOffset ; 
+      double sideYPos = sideYOffset ;   
+      double sideZPos = 0.0; 
+      double sideXPos2 = sideXOffset2 ; 
+      double sideYPos2 = sideYOffset2 ;       
+
+      dd4hep::Position sideTrans;
+      dd4hep::PlacedVolume sidePhys;
+      dd4hep::DetElement sideDE;
+
+      if (side % 2 == 0) {
```

Comment:
It feels like this can be much simplified. Many calculations are done and then never used depending on `side % 2`